### DISCUSSION TITLE
Webservice returns me when retrieving addresses: "Error downloading the requested resource"

### DISCUSSION THREAD

**Discussion by JaimeSistemas** [osm-search/Nominatim#3431] [Q&A] [answered]
The "Error downloading the requested resource" happened to me again since yesterday, I cannot consume the websersive. Last time it was because of the timeouts that made the queries slow but this time I don't know what I can do. 
I hope you can help.

## What result did you expect?

**When the result in the right place and just named wrongly:** 

**When the result missing completely:**

**Comment by mtmail** [comment]
Are you setting a user agent string https://github.com/osm-search/Nominatim/issues/3405#issuecomment-2086532576 and follow the other requirements of the https://operations.osmfoundation.org/policies/nominatim/ ?

**Reply by JaimeSistemas** [reply]
> Are you setting a user agent string [#3405 (comment)](https://github.com/osm-search/Nominatim/issues/3405#issuecomment-2086532576) and follow the other requirements of the https://operations.osmfoundation.org/policies/nominatim/ ?

![errordirecciones](https://github.com/osm-search/Nominatim/assets/168585107/842eefcd-de6e-47e7-bd41-657d04ee3c5f)

Before the incident, the webservices had been consuming it like this without any problem. I attach the image of the source code

**Reply by mtmail** [reply]
You didn't answer the questions.

**Reply by JaimeSistemas** [reply]
> You didn't answer the questions.

I am not using a User Agent, I only consume HTTP Referer to extract the address

**Reply by mtmail** [reply]
Best if you change the app to provide a valid HTTP Referer or User-Agent identifying the application. Then post the string here and a server administrator can look at the requests the app is sending.

**Reply by JaimeSistemas** [reply]
It returns this error code -2146697208. I am interested in what you tell me, how do I validate HTTP Referer or User-Agent. I'm interested in what you tell me, how do I validate HTTP Referer or User-Agent. I am new to using this platform

> Best if you change the app to provide a valid HTTP Referer or User-Agent identifying the application. Then post the string here and a server administrator can look at the requests the app is sending.

**Reply by mtmail** [reply]
What platform are you using? It looks like Visual Basic Script inside MS Office/Excel.

**Reply by JaimeSistemas** [reply]
> What platform are you using? It looks like Visual Basic Script inside MS Office/Excel.

My project is made with visual basic 6. Should I add something to the code?

**Reply by mtmail** [reply]
You need to find out how to set a User Agent string. A Visual Basic forum might be a better place to ask.

https://codingislove.com/http-requests-excel-vba/ suggests it's something like `xmlhttp.setRequestHeader "User-Agent", "name-of-your-app"` 

https://codingislove.com/weather-app-in-excel-vba/ explains how to call `xmlhttp` and convert the response to a `DOMDocument`

**Reply by JaimeSistemas** [reply, marked as answer]
> You need to find out how to set a User Agent string. A Visual Basic forum might be a better place to ask.
> 
> https://codingislove.com/http-requests-excel-vba/ suggests it's something like `xmlhttp.setRequestHeader "User-Agent", "name-of-your-app"`
> 
> https://codingislove.com/weather-app-in-excel-vba/ explains how to call `xmlhttp` and convert the response to a `DOMDocument`

the information from the user agent and the service responded to me with information, your advice was helpful to me. Thank you very much for your help you saved me
